joern-cli/frontends/rubysrc2cpg/src/main/scala/io/joern/rubysrc2cpg/passes/AstPackagePass.scala

```diff
@@ -0,0 +1,37 @@
+package io.joern.rubysrc2cpg.passes
+
+import io.joern.rubysrc2cpg.astcreation.AstCreator
+import io.joern.rubysrc2cpg.utils.{PackageContext, PackageTable}
+import io.joern.x2cpg.datastructures.Global
+import io.shiftleft.codepropertygraph.Cpg
+import io.shiftleft.passes.ConcurrentWriterCpgPass
+import better.files.File
+
+import scala.util.Try
+
+class AstPackagePass(cpg: Cpg, tempExtDir: String, global: Global, packageTable: PackageTable, inputPath: String)
+    extends ConcurrentWriterCpgPass[String](cpg) {
+  override def generateParts(): Array[String] = getRubyDependenciesFile(inputPath) ++ getRubyDependenciesFile(tempExtDir)
+
+  override def runOnPart(diffGraph: DiffGraphBuilder, filePath: String): Unit = {
+    Try(new AstCreator(filePath, global, PackageContext(resolveModuleNameFromPath(filePath), packageTable)).createAst())
+  }
+
+  private def getRubyDependenciesFile(inputPath: String): Array[String] = {
+    val currentDir = File(inputPath)
+    if (currentDir.exists) {
+      currentDir.listRecursively.filter(_.extension.exists(_ == ".rb")).map(_.path.toString).toArray
+    } else {
+      Array.empty
+    }
+  }
+
+  private def resolveModuleNameFromPath(path: String): String = {
+    if (path.contains(tempExtDir)) {
+      val moduleNameRegex = s"${java.io.File.separator}gems${java.io.File.separator}([^${java.io.File.separator}]+)${java.io.File.separator}lib${java.io.File.separator}.*".r
```

Comment:
Does this work? This should de-duplicate the number of times `java.io.File.separator` is written
```suggestion
      val moduleNameRegex = Seq("gems", "([^", "]+)", "lib", ".*").mkString(java.io.File.separator).r
```